src/lib.rs

```diff
@@ -31,25 +36,41 @@ mod inner {
         const BLOCK_SIZE: u8 = 8;
 
         #[new]
-        #[pyo3(signature = (string = Vec::new()))]
-        fn new(string: Vec<u8>) -> Self {
-            SeaHash { inner: string }
+        #[pyo3(signature = (data = Vec::new()))]
+        fn new(data: Vec<u8>) -> Self {
+            let mut inner = seahash::SeaHasher::new();
+            inner.write(data.as_slice());
+            SeaHash { inner }
         }
 
+        #[cfg(not(Py_3_11))]
         pub fn update(&mut self, obj: &[u8]) {
-            self.inner.extend_from_slice(obj)
+            self.inner.write(obj)
+        }
+
+        #[cfg(Py_3_11)]
+        pub fn update(&mut self, py: Python, obj: &PyAny) -> PyResult<()> {
+            match obj.extract() {
+                Ok(buf) => {
+                    self.inner.write(buf);
+                }
+                Err(_) => {
+                    self.inner.write(PyBuffer::get(obj)?.to_vec(py)?.as_slice());
+                }
+            }
+            Ok(())
         }
 
         pub fn digest(&self, py: Python) -> PyObject {
-            PyBytes::new(py, &self.intdigest(py).to_ne_bytes()).into()
+            PyBytes::new(py, &self.intdigest().to_ne_bytes()).into()
         }
 
-        pub fn intdigest(&self, py: Python) -> u64 {
-            hash(py, &self.inner)
+        pub fn intdigest(&self) -> u64 {
+            self.inner.finish()
```

Comment:
Suggestion: Given `finish` likely computes the final hash, should it unlock the GIL?

Reply:
The hash gets updated every time bytes are written to the hasher using `write` (which calls out to [`push_bytes`](https://docs.rs/seahash/latest/src/seahash/stream.rs.html#59-157)). Technically, `update` would gain more by unlocking the GIL. [`finish`](https://docs.rs/seahash/latest/src/seahash/stream.rs.html#166-175) only computes the hash of the tail if there is one.